### PR TITLE
`writeData()` force evaluation of `x` before options are set

### LINKED ISSUE #264: writeData converts character column to numeric value
**Describe the bug**

Function "base::format" is used to convert a numeric value 1234.1 to a character "1234.1" in a data.frame using dplyr , writeData writes the string "1234.0999999999999". I don't fully understand what's happening. Maybe this is a problem of lazy evaluation.

The bug doesn't appear, if I use base::paste. 

The bug doesn't appear, if the data.frame is bound to a variable and then written 

**To Reproduce**

```
library(openxlsx)
library(magrittr)
library(dplyr)
library(XML)

fileName <- "test.xlsx"
wb <- createWorkbook()
sheet <- addWorksheet(wb,"Sheet")
data.frame(a=1234.1) %>% mutate(a=format(a)) %>% writeData(wb,sheet,.)
saveWorkbook(wb,fileName,overwrite=T)
unzip(fileName,exdir="tmp")
XML::xmlParse("tmp/xl/sharedStrings.xml")
```

Output

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<sst xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main" count="2" uniqueCount="2">
  <si>
    <t xml:space="preserve">a</t>
  </si>
  <si>
    <t xml:space="preserve">1234.0999999999999</t>
  </si>
</sst>
```

**Expected behavior**
String "1234.1" instead of a weird casted floating point number.

**Additional context**

This works:

```
library(openxlsx)
library(magrittr)
library(dplyr)
library(XML)

fileName <- "test.xlsx"
wb <- createWorkbook()
sheet <- addWorksheet(wb,"Sheet")
data.frame(a=1234.1) %>% mutate(a=paste(a)) %>% writeData(wb,sheet,.)
saveWorkbook(wb,fileName,overwrite=T)
unzip(fileName,exdir="tmp")
XML::xmlParse("tmp/xl/sharedStrings.xml")
```

This also works:

```
library(openxlsx)
library(magrittr)
library(dplyr)
library(XML)

fileName <- "test.xlsx"
wb <- createWorkbook()
sheet <- addWorksheet(wb,"Sheet")
df <- data.frame(a=1234.1) %>% mutate(a=format(a))
df %>% writeData(wb,sheet,.)
saveWorkbook(wb,fileName,overwrite=T)
unzip(fileName,exdir="tmp")
XML::xmlParse("tmp/xl/sharedStrings.xml")
```

### PR DESCRIPTION
**Pull request by jmbarbone**
Simply using `force(x)` before options seems to do the trick.

No local test fails.

resolves #264

### CODE REVIEW

File: tests/testthat/test-writeData.R [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  writeData(wb, "sheet", startCol = 2, data.frame(b = as.character(123.4)))
+  writeData(wb, "sheet", startCol = 3, data.frame(c = "123.4"))
+  writeData(wb, "sheet", startCol = 4, df)
+  writeData(wb, "sheet", startCol = 5, df2)
+  
+  saveWorkbook(wb, wbfile)
+  out <- read.xlsx(wbfile)
+  
+  # Possibly overkill
+
```
Comment (JanMarvin) [inline tests/testthat/test-writeData.R:21]: 🤣 Just my thoughts

File: NEWS.md [hunk 2]
```diff
@@ -9,12 +9,11 @@
 * add option to save as read only recommended ([#201](https://github.com/ychps/openxlsx/issues/201))
 * fixed writing hyperlink formulas ([#200](https://github.com/ychps/openxlsx/issues/200))
 * `write.xlsx()` now throws an error if it doesn't have write permissions ([#190](https://github.com/ycphs/openxlsx/issues/190))
-* `write.xlsx()` now again uses the default of `overwrite = TRUE` for saving files ([#249](https://github.com/ycphs/openxlsx/issues/249))
```
Comment (JanMarvin) [inline NEWS.md]: why did you remove this?

**Comment by JanMarvin** [comment]
Thanks, simple and elegant! LGTM!

File: NEWS.md (hunk 2 above)
Comment (jmbarbone) [reply NEWS.md]: By mistake probably.  Didn't notice this.

**Review by JanMarvin — APPROVED** [review]

**Comment by JanMarvin** [comment]
could you please rebase on/or merge with master?

**Comment by JanMarvin** [comment]
merged in d2f7e03a54dd613d13a6b8b02d39df7f59ce725c , thanks
